react-form-controls/my-app/src/RegistrationFormControlled.tsx

```diff
@@ -0,0 +1,43 @@
+import { useState, type FormEvent } from 'react';
+
+type Props = {
+  name?: string;
+  pwd?: string;
+};
+
+export function RegistrationFormControlled({ pwd, name }: Props) {
+  const [username, setUsername] = useState(name);
+  const [password, setPassword] = useState(pwd);
+
+  function handleSubmit(event: FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+    console.log(event.target);
+    console.log(event.currentTarget);
+    const form = new FormData(event.currentTarget);
```

Comment:
In a controlled form, you have access to the values in `username` and `password`, so you don't need to extract them from `FormData`.